File: C++/hsScalarTriple.cpp
#include "hsScalarTriple.h"

hsScalarTriple::hsScalarTriple() : X(0.0f), Y(0.0f), Z(0.0f) { }

hsScalarTriple::hsScalarTriple(float x, float y, float z) {
    X = x;
    Y = y;
    Z = z;
}

hsScalarTriple::~hsScalarTriple() { }

float hsScalarTriple::getX() { return X; }
float hsScalarTriple::getY() { return Y; }
float hsScalarTriple::getZ() { return Z; }
void hsScalarTriple::setX(float x) { X = x; }
void hsScalarTriple::setY(float y) { Y = y; }
void hsScalarTriple::setZ(float z) { Z = z; }

void hsScalarTriple::read(hsStream *S) {
    X = S->readFloat();
    Y = S->readFloat();
    Z = S->readFloat();
}

void hsScalarTriple::write(hsStream *S) {
    S->writeFloat(X);
    S->writeFloat(Y);
    S->writeFloat(Z);
}

File: Java/Plasma/hsScalarTriple.java
package Plasma;

import java.io.IOException;
import Plasma.CoreLib.hsStream;

public class hsScalarTriple {
    private float a, b, c;
    
    public hsScalarTriple() {
        this(0.0f, 0.0f, 0.0f);
    }
    public hsScalarTriple(float f1, float f2, float f3) {
        a = f1;
        b = f2;
        c = f3;
    }
    
    public void read(hsStream S) throws IOException {
        a = S.readFloat();
        b = S.readFloat();
        c = S.readFloat();
    }
    
    public void write(hsStream S) throws IOException {
        S.writeFloat(a);
        S.writeFloat(b);
        S.writeFloat(c);
    }
}

